### PR TITLE
python312Packages.facedancer: 2019.3.2 -> 3.0.4

### PR DESCRIPTION
**Pull request by carlossless**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by carlossless** [comment]
Result of `nixpkgs-review pr 353404` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

**Comment by sarahec** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353404`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/facedancer/default.nix [hunk 1]
```diff
@@ -2,36 +2,53 @@
...
 
 buildPythonPackage rec {
   pname = "facedancer";
-  version = "2019.3.2";
-  format = "setuptools";
+  version = "3.0.4";
+  pyproject = true;
+  disabled = pythonOlder "3.8";
 
   src = fetchPypi {
```
Comment (sarahec) [inline pkgs/development/python-modules/facedancer/default.nix:19]: It's better to build this from source, see https://github.com/greatscottgadgets/facedancer

File: pkgs/development/python-modules/facedancer/default.nix [hunk 2]
```diff
@@ -2,36 +2,53 @@
...
+  ];
 
-  meta = with lib; {
-    description = "library for emulating usb devices";
-    homepage = "https://greatscottgadgets.com/greatfet/";
-    license = licenses.bsd3;
-    maintainers = with maintainers; [ mog ];
+  meta = {
+    description = "A library for implementing your own USB device in Python, supported by a hardware peripheral such as Cynthion or GreatFET";
+    homepage = "https://github.com/greatscottgadgets/facedancer";
```
Comment (sarahec) [inline pkgs/development/python-modules/facedancer/default.nix:47]: Homepage is good, please add a changelog

**Review by sarahec — CHANGES_REQUESTED** [review]

**Comment by sarahec** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353404`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 353404` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

**Review by sarahec — APPROVED** [review]
Thank you. LGTM.

**Comment by carlossless** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353404`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.facedancer</li>
    <li>python311Packages.facedancer.dist</li>
    <li>python312Packages.facedancer</li>
    <li>python312Packages.facedancer.dist</li>
  </ul>
</details>

**Pull request merged**